### PR TITLE
Dockerfile and docker-compose

### PR DESCRIPTION
**Pull request by angelbarrera92**
Working version solving issue #1 
With those files, you can run the app locally using sqlite *(as you use locally)*.
```
docker build -t trade-motors:0.1.0 .
docker run -d -p 8000:8000 trade-motors:0.1.0
```
Or you can use `docker-compose.yml` file which will run a mysql server along with the web app. 
```
docker-compose up
```

### CODE REVIEW

**Comment by haroon-sheikh** [comment]
Thanks @angelbarrera92 for the PR. Do you think it's a good idea to use alpine version of docker images instead? i.e. python:2.7-alpine

File: src/gp_cars/settings/docker.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.mysql',
+        'HOST': 'db',                      
+        'PORT': '3306', 
+        'NAME': 'trademotors',
+        'USER': 'trademotors',
+        'PASSWORD': 'sitture2',
+    }
```
Comment (haroon-sheikh) [inline src/gp_cars/settings/docker.py]: Do these now need to come from environment variables?

Comment (angelbarrera92) [reply src/gp_cars/settings/docker.py]: I can add `RUN python manage.py test` on Dockerfile. This way the image will not be built if there are test failing.
Yes, you are right. The DB Name, user and password should go on environment variables.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+version: "2.1"
+services:
+  db:
+    image: mysql/mysql-server:5.7
+    environment:
+      - MYSQL_DATABASE=trademotors
+      - MYSQL_USER=trademotors
+      - MYSQL_PASSWORD=sitture2
+      - MYSQL_ROOT_HOST=0.0.0.0
```
Comment (haroon-sheikh) [inline docker-compose.yml]: how about if we store the shared variables in a `.env` file and reference in the file. E.g. https://docs.docker.com/compose/environment-variables/#setting-environment-variables-in-containers

**Comment by angelbarrera92** [comment]
The DB parameters have been externalized and I also added a test command on image building process.

File: docker-compose.yml (hunk 2 above)
Comment (angelbarrera92) [reply docker-compose.yml]: Sounds Good!

**Comment by angelbarrera92** [comment]
Added `.env` file and modified `docker-compose.yml`

About alpine tag... I have some troubles with the mysql client and I think is better to use debian base image during development.

**Comment by haroon-sheikh** [comment]
it looks good to merge now. Do you want me to merge this now or wait until tomorrow?

**Comment by haroon-sheikh** [comment]
@angelbarrera92 can you add a section in README.md about building/running with Docker?

**Comment by angelbarrera92** [comment]
@haroon-sheikh I will add docker section in README.md. I will do a push ASAP.

**Comment by angelbarrera92** [comment]
Done. Docker info in README.md

**Comment by angelbarrera92** [comment]
@haroon-sheikh it's all ok?

**Pull request merged**